presto-router/src/test/java/com/facebook/presto/router/TestCustomSchedulerManager.java

```diff
@@ -0,0 +1,168 @@
+        String schedulerName = "router-scheduler.name=mock";
+        Files.write(tempPluginSchedulerConfigFile, schedulerName.getBytes());
+
+        SchedulerFactory mockSchedulerFactory = new MockSchedulerFactory();
+        CustomSchedulerManager customSchedulerManager = new CustomSchedulerManager(ImmutableList.of(mockSchedulerFactory), tempPluginSchedulerConfigFile.toFile());
+
+        Bootstrap app = new Bootstrap(
+                new TestingNodeModule("test"),
+                new TestingHttpServerModule(),
+                new JsonModule(),
+                new JaxrsModule(true),
+                new RouterModule(Optional.of(customSchedulerManager)));
+
+        Injector injector = app.doNotInitializeLogging()
+                .setRequiredConfigurationProperty("router.config-file", configFile.getAbsolutePath())
+                .quiet().initialize();
+
+        schedulerManager = injector.getInstance(CustomSchedulerManager.class);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+            throws Exception
+    {
+        for (TestingPrestoServer prestoServer : prestoServers) {
+            prestoServer.close();
+        }
+    }
+
+    static class MockSchedulerFactory
+            implements SchedulerFactory
+    {
+        public static final String MOCK = "mock";
+
+        @Override
+        public String getName()
+        {
+            return MOCK;
+        }
+
+        @Override
+        public Scheduler create(Map<String, String> config)
+        {
+            return new MockScheduler();
+        }
+    }
+
+    public static class MockScheduler
+            implements Scheduler
+    {
+        @Override
+        public Optional<URI> getDestination(String user)
+        {
+            return Optional.empty();
+        }
+
+        @Override
+        public void setCandidates(List<URI> candidates) {}
+    }
+
+    public static File getConfigFile(List<TestingPrestoServer> servers, File tempFile)
+            throws IOException
+    {
+        List<URI> serverURIs = servers.stream()
+                .map(TestingPrestoServer::getBaseUrl)
+                .collect(Collectors.toList());
+        RouterSpec spec = new RouterSpec(ImmutableList.of(new GroupSpec("all", serverURIs, Optional.empty(), Optional.empty())),
+                ImmutableList.of(new SelectorRuleSpec(Optional.empty(), Optional.empty(), Optional.empty(), "all")),
+                Optional.of(CUSTOM_PLUGIN_SCHEDULER),
+                Optional.empty());
+        JsonCodec<RouterSpec> codec = jsonCodec(RouterSpec.class);
+        Files.write(tempFile.toPath(), codec.toBytes(spec));
+        return tempFile;
+    }
+
+    @Test
+    public void testCustomSchedulerCreation()
+            throws InterruptedException
+    {
+        schedulerManager.loadScheduler();
+        assertEquals(schedulerManager.getScheduler().getClass(), MockScheduler.class);
```

Comment:
Let's add an assertion that proves that the MockScheduler is called when a request is made to the router -

1. Add a `int requestsMade = 0` field in `MockScheduler`. Increment `requestMade` in `MockScheduler#getDestination`
1. Route a SQL execution to the router started up in the test setup, similar to how does it`TestClusterManager#testQuery`
1. The call will fail (catch the exception); but `schedulerManager.getScheduler().getRequestsMade()` should return a non-zero value